### PR TITLE
Remove testing module.

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Remove the testing module.

## Motivation and Context

See https://github.com/glotzerlab/signac/issues/709

## Checklist:

- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] The [package documentation](https://github.com/glotzerlab/signac-flow/tree/master/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -15,6 +15,11 @@ Fixed
 
 - Detecting correct environment on Delta GPU nodes (#682).
 
+Removed
++++++++
+
+- The ``signac.testing`` module (#691).
```
Comment (bdice) [inline changelog.txt:21]:
```suggestion
- The ``flow.testing`` module (#691).
```

File: flow/testing.py [hunk 2]
```diff
@@ -1,36 +0,0 @@
...
-    \*\*kwargs
-        Keyword arguments forwarded to :meth:`signac.testing.init_jobs`.
-
-    Returns
-    -------
-    :class:`signac.Project`
-        Project with initialized jobs.
-
-    """
-    init(alias=alias, root=root, template="testing")
```
Comment (bdice) [inline flow/testing.py:33]: A next step is to determine if the “testing” template is worth keeping or not. That can be a separate PR if desired.

**Review by bdice — APPROVED** [review]
One small fix to the changelog, otherwise LGTM.

File: flow/testing.py (hunk 2 above)
Comment (vyasr) [reply flow/testing.py:33]: Agreed, we can consider that too. Opened #692
